api/v1beta1/conversion.go

```diff
@@ -16,14 +16,187 @@ limitations under the License.
 
 package v1beta1
 
-// Hub marks CloudStackCluster as a conversion hub.
-func (*CloudStackCluster) Hub() {}
+import (
+	"context"
+	"fmt"
+	"github.com/apache/cloudstack-go/v2/cloudstack"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	conv "k8s.io/apimachinery/pkg/conversion"
+	"sigs.k8s.io/cluster-api-provider-cloudstack/api/v1beta2"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
 
-// Hub marks CloudStackClusterList as a conversion hub.
-func (*CloudStackClusterList) Hub() {}
+const DefaultEndpointCredential = "global"
 
-// Hub marks CloudStackMachine as a conversion hub.
-func (*CloudStackMachine) Hub() {}
+//nolint:golint,revive,stylecheck
+func Convert_v1beta1_CloudStackCluster_To_v1beta2_CloudStackCluster(in *CloudStackCluster, out *v1beta2.CloudStackCluster, s conv.Scope) error {
+	out.ObjectMeta = in.ObjectMeta
+	failureDomains, err := getFailureDomains(in)
+	if err != nil {
+		return err
+	}
+	out.Spec = v1beta2.CloudStackClusterSpec{
+		ControlPlaneEndpoint: in.Spec.ControlPlaneEndpoint,
+		FailureDomains:       failureDomains,
+		Account:              in.Spec.Account,
+		Domain:               in.Spec.Domain,
```

Comment:
why are we setting these attributes for v1beta2? Should those be captured inside the failuredomain?

Reply:
Somehow v1beta2 cluste spec still supports Account and Domain outside of the failure domains. We might need to discuss about removing these with @rejoshed

Reply:
failuredomain has domain/account.

Keeping them here for backward compatible. Can be removed once v1beta1 backward compatible is needed.

Reply:
It seems redundant. They're in two places but it feels like they should only be in the failuredomains. Then if there are any inconsistencies in account/domain between the failure domains, conversion should fail